src/test/regress/bin/normalize.sed

```diff
@@ -226,7 +226,7 @@ s/ERROR:  ROLLBACK is not allowed in an SQL function/ERROR:  ROLLBACK is not all
 /Parent Relationship/d
 /Parent-Relationship/d
 s/function array_cat_agg\(anyarray\) anyarray/function array_cat_agg\(anycompatiblearray\) anycompatiblearray/g
-s/function array_cat_agg\(anycompatiblearray\)/function array_cat_agg\(anyarray\)/g
+s/function array_cat_agg\(anyarray\)/function array_cat_agg\(anycompatiblearray\)/g
```

Comment:
I don't really understand this change

Reply:
Makes sense, I am adding a small description to the commit.

Starting with PG14, `array_cat_agg` now needs to take `anycompatiblearray` instead of `anyarray` because `array_cat` changed its type from `anyarray` to `anycompatiblearray` with PG14.

We have two problematic line prints with PG13 in multi_extension.out and upgrade_list_citus_objects.out:
 1. `function array_cat_agg(anyarray) anyarray`
 2. `function array_cat_agg(anyarray)`

PG14+ these lines are:
 1. `function array_cat_agg(anycompatiblearray) anycompatiblearray`
 2. `function array_cat_agg(anycompatiblearray)`

The previous state of the normalize file was the following, and was actually wrong because the second line messes up with the first:
```
s/function array_cat_agg\(anyarray\) anyarray/function array_cat_agg\(anycompatiblearray\) anycompatiblearray/g
s/function array_cat_agg\(anycompatiblearray\)/function array_cat_agg\(anyarray\)/g
```
With that previous state, we end up with:
 1. `function array_cat_agg(anyarray) anycompatiblearray` this is logically not correct
 2. `function array_cat_agg(anyarray)` this is okay, but I'd rather keep the PG14+ print since PG13 will be dropped soon.

HENCE, I made the change.